src/icons/fileIcons.ts

```diff
@@ -2409,5 +2409,10 @@ export const fileIcons: FileIcons = {
     },
     { name: 'roblox', fileExtensions: ['rbxl', 'rbxlx', 'rbxm', 'rbxmx'] },
     { name: 'panda', fileNames: ['panda.config.ts', 'panda.config.js'] },
+    {
```

Comment:
In my opinion the file names and file extensions are too generic. This means that they may not only be related for Minecraft but also for other projects as well. For example the other minecraft icon is only used for file names beginning with "mc" which is more related than just "mixins.jsoö". Hopefully you got my point here :) 

There's no official support for this, but you could try out to customize this extension to use your custom icons as described here:
https://github.com/PKief/vscode-material-icon-theme#custom-svg-folder-icons

Reply:
I researched if `*.mixins.json` was being used in scenarios other than forge and didn't find much, but I agree that it's pretty general. I can remove the extension and keep the file name. Let me know.